### PR TITLE
feat: GitOps Operator enables RHSSO for each ArgoCD Instance(#GITOPS-946)

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind enhancement

**What does this PR do / why we need it**:
GitOps Operator enables RHSSO for each ArgoCD Instance
* register OpenShift APIs: apps, oauth, and templates
* add "sso"  parameters in CSV
* add permission to gitops-operator sa in CSV: oauthclients, template
* add SSO config to NewCR in the default instance
* pickup SSO changes from upstream
* As keycloak will be a default workload, increase the existing resource limits and requests on the openshift-gitops namespace.

**Have you updated the necessary documentation?**
NA

**Which issue(s) this PR fixes**:
GitOps Operator enables RHSSO for each ArgoCD Instance

Fixes #?
GitOps Operator enables RHSSO for each ArgoCD Instance

**Test acceptance criteria**:

* [x] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:
Create Operator and verify RHSSO on the default ArgoCD instance

### CODE REVIEW

File: pkg/controller/gitopsservice/gitopsservice_controller.go [hunk 1]
```diff
@@ -252,10 +252,13 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
...
 				Hard: corev1.ResourceList{
-					corev1.ResourceMemory:       resourcev1.MustParse("4032Mi"),
-					corev1.ResourceCPU:          resourcev1.MustParse("6188m"),
-					corev1.ResourceLimitsMemory: resourcev1.MustParse("8046Mi"),
-					corev1.ResourceLimitsCPU:    resourcev1.MustParse("12750m"),
+					corev1.ResourceMemory:       resourcev1.MustParse("4544Mi"),
+					corev1.ResourceCPU:          resourcev1.MustParse("6688m"),
+					corev1.ResourceLimitsMemory: resourcev1.MustParse("9070Mi"),
+					corev1.ResourceLimitsCPU:    resourcev1.MustParse("13750m"),
+				},
```
Comment (wtam2018) [inline pkg/controller/gitopsservice/gitopsservice_controller.go:259]: Just wondering why these are changed in this PR.

Comment (iam-veeramalla) [reply pkg/controller/gitopsservice/gitopsservice_controller.go:259]: As keycloak will be a default workload from v1.2, we have to increase the existing resource limits and requests on the openshift-gitops namespace.

File: deploy/argocd_role.yaml [hunk 2]
```diff
@@ -70,13 +82,48 @@ rules:
   - servicemonitors
   verbs:
   - '*'
+- apiGroups:
+  - ""
+  resources:
+  - resourcequotas
+  verbs:
+  - 'list'
```
Comment (wtam2018) [inline deploy/argocd_role.yaml:90]: please remove single quotes.

File: deploy/olm-catalog/gitops-operator/manifests/gitops-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -239,10 +236,8 @@ spec:
                 - name: OPERATOR_NAME
                   value: gitops-operator
                 - name: DISABLE_DEX
-                  value: "true"
-                - name: ARGOCD_REDIS_IMAGE
-                  value: "registry.redhat.io/rhel8/redis-5"
-                image: quay.io/redhat-developer/gitops-backend-operator:v0.0.3
+                  value: "true"                  
+                image:  registry-proxy.engineering.redhat.com/rh-osbs/openshift-gitops-1-gitops-rhel8-operator:v1.2.0
```
Comment (wtam2018) [inline deploy/olm-catalog/gitops-operator/manifests/gitops-operator.clusterserviceversion.yaml:240]:
```suggestion
                image: quay.io/redhat-developer/gitops-backend-operator:v0.0.3
```

File: deploy/olm-catalog/gitops-operator/manifests/gitops-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -334,6 +320,53 @@ spec:
...
+          - watch
+        - apiGroups:
+          - ""
+          resources:
+          - resourcequotas
+          verbs:
+          - 'list'
+          - 'get'
+          - 'create'
+          - 'watch'
```
Comment (wtam2018) [inline deploy/olm-catalog/gitops-operator/manifests/gitops-operator.clusterserviceversion.yaml:357]:
```suggestion
          - list
          - get
          - create
          - watch
```

File: deploy/argocd_role.yaml [hunk 5]
```diff
@@ -70,13 +82,48 @@ rules:
...
+  - 'get'
+  - 'create'
+  - 'watch'
```
Comment (wtam2018) [inline deploy/argocd_role.yaml:93]:
```suggestion
  - list
  - get
  - create
  - watch
```

**Review by wtam2018 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**